Frontend/src/components/TaskList.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { ITask } from '../interfaces/ITask';
+
+export async function fetchTasks(): Promise<ITask[]> {
+  const taskListResp = await axios.get('list_tasks');
+  return taskListResp.data.tasks;
+}
+
+export default function TaskList() {
+  const [tasks, setTasks] = useState<ITask[]>([]);
+
+  useEffect(() => {
+    const taskHandler = async () => {
+      const taskList = await fetchTasks();
+      setTasks(taskList);
+    };
+    taskHandler();
+  }, []);
+
+  return (
+    <div>
+      <div>
+        <h1>Pending</h1>
+        <ul>
+          {tasks
+            .filter((task) => task.isCompleted === false)
+            .map((task) => {
+              return (
+                <li key={task.id}>
+                  {task.id}) {task.title}
+                </li>
+              );
+            })}
+        </ul>
```

Comment:
This code works, however, there is a repetition here that could be removed. 

the highlighted code and and lines 39-49 only differ by one word - the boolean. Please extract that shared logic to a new component and reuse it in place of lines 25-35 and 39-49.

Reply:
Done. I was having problems with the linter and TS, but I figured it out.

Reply:
@mEWeil would love to see these changes when you get the chance to push them up :)

Reply:
I'll have it pushed pretty soon.